src/optimize/dynamic_dll_plugin/dll_compiler.js

```diff
@@ -197,18 +217,65 @@ export class DllCompiler {
         }));
 
         if (webpackErrors) {
-          this.logWithMetadata(
-            ['fatal', 'optimize:dynamic_dll_plugin'],
-            `Client vendors dll compilation failed`
-          );
+          // Reject with webpack fatal errors
           return reject(webpackErrors);
         }
 
-        // Otherwise let it proceed
-        this.logWithMetadata(
-          ['info', 'optimize:dynamic_dll_plugin'],
-          `Client vendors dll compilation finished with success`
-        );
+        // Identify if we have not allowed modules
+        // bundled inside the dll bundle
+        const notAllowedModules = [];
+
+        stats.compilation.modules.forEach((module) => {
+          // ignore if no module or userRequest are defined
+          if (!module || !module.resource) {
+            return;
+          }
+
+          // ignore if this module represents the
+          // dll entry file
+          if (module.resource === this.getEntryPath()) {
+            return;
+          }
+
+          // ignore if this module is part of the
+          // files inside dynamic dll plugin public folder
+          if (inDllPluginPublic(module.resource)) {
+            return;
+          }
+
+          // A module is not allowed if it's not a node_module, a webpackShim
+          // or the reasons from being bundled into the dll are not node_modules
```

Comment:
Out of curiosity, what's an example of a module that's not a node_module or a webpackShim, but is being bundled into the dll because of a require statement from `node_module`s?

Reply:
I feel like such situations are _probably_ issues that we wouldn't want to ignore.

Reply:
For example right now we have the example of `elastic/datemath` being imported by `@elastic/eui` (which is a node_module) and for this case we need both bundled by the dll